homeassistant/bootstrap.py

```diff
@@ -73,6 +73,63 @@ def _handle_requirements(hass: core.HomeAssistant, component,
     return True
 
 
+def get_validated_component_config(hass: core.HomeAssistant, domain: str,
+                                   config) -> bool:
+    """Validate config for a domain, including all platforms."""
+    component = loader.get_component(domain)
+    missing_deps = [dep for dep in getattr(component, 'DEPENDENCIES', [])
+                    if dep not in hass.config.components]
+
+    if missing_deps:
+        _LOGGER.error(
+            'Not initializing %s because not all dependencies loaded: %s',
+            domain, ", ".join(missing_deps))
+        return False
```

Comment:
If we return something, let's return None. However, I feel like maybe raising an exception would be more appropriate ?